cmd/soci/commands/index/info.go

```diff
@@ -52,6 +62,20 @@ var infoCommand = cli.Command{
 
 		_, err = io.Copy(os.Stdout, reader)
 		return err
-
 	},
 }
+
+func isZtocDigest(digest string) (bool, error) {
+	db, err := soci.NewDB()
+	if err != nil {
+		return false, err
+	}
+	ae, err := db.GetArtifactEntry(digest)
+	if err != nil {
+		return false, err
+	}
+	if ae.Type == soci.ArtifactEntryTypeLayer {
+		return true, nil
+	}
```

Comment:
why not check if `ae.Type == soci.ArtifactEntryTypeIndex` and only proceed if it's a soci index? Or just let it return `ArtifactEntryType` so it can be reused by `ztoc` (e.g., if we only want to get ztoc not soci index)

Reply:
Because it would be nice to redirect users to ``soci ztoc`` command to get detailed info of ztoc, if the input is ztoc digest, which is what is proposed in the description of the ticket #258

Reply:
We should show the error message to redirect users to `soci ztoc`, and also make this function resuable. Otherwise we'll have a almost identical function for `isIndexDigest`.`soci ztoc info` has a similar issue:

```shell
$ soci index list
DIGEST                                                                     SIZE    IMAGE REF                          PLATFORM       CREATED
sha256:e0ad85cffcf93aba3e1b75cb7297e1c8995b4714d966354be56b8fbbef98af6c    731     docker.io/library/ubuntu:latest    linux/amd64    95h36m35s ago
$ soci ztoc info sha256:e0ad85cffcf93aba3e1b75cb7297e1c8995b4714d966354be56b8fbbef98af6c
soci: cannot unmarshal ztoc: runtime error: slice bounds out of range [1701651067:731]
```

Suggest to name it `checkSociDigestType` and return the type, or `validateSociDigestType` and pass a digest and expected type.

Reply:
We _are_ showing the error msg of redirecting users to use ``soci ztoc info``. Please check line 49.

Reply:
Yes, I mean "I agree we should show the error message to redirect users to soci ztoc (your line 49), **but also make this function resuable**" :)

Reply:
Added ``GetArtifactType`` method in ``soci/artifacts.go``